File: dotnet-backend/Dtos/RecommendationDtos.cs
namespace Wellness.Backup.Api.Dtos;

/// <summary>
/// DTOs for recommendation endpoints mirrored from Spring Boot.
/// </summary>
/// <remarks>@author Tiong Zhong Cheng</remarks>
public sealed record RecommendationResponse(
    long? Id,
    string Title,
    string TrendSummary,
    string RecommendationText,
    IReadOnlyList<string> ActionItems,
    string GeneratedBy,
    DateTime? CreatedAt);

public sealed record InternalRecommendationRequest(
    string? Title,
    string? TrendSummary,
    string? RecommendationText,
    IReadOnlyList<string>? ActionItems,
    string? GeneratedBy);

File: spring-backend/src/main/java/sg/edu/nus/iss/wellness/dto/RecommendationDtos.java
package sg.edu.nus.iss.wellness.dto;

import java.time.Instant;
import java.util.List;

/**
 * DTOs for recommendation endpoints.
 *
 * @author Tiong Zhong Cheng
 */
public final class RecommendationDtos {
    private RecommendationDtos() {
    }

    public record RecommendationResponse(
            Long id,
            String title,
            String trendSummary,
            String recommendationText,
            List<String> actionItems,
            String generatedBy,
            Instant createdAt
    ) {
    }

    public record InternalRecommendationRequest(
            String title,
            String trendSummary,
            String recommendationText,
            List<String> actionItems,
            String generatedBy
    ) {
    }
}

